File: ToDo/ToDo.Web/Modules/Administration/Movie/MovieForm.cs
using MovieTutorial.MovieDB;
using Serenity.ComponentModel;
using System;
//using ToDo.Modules.Administration.Movie;

namespace ToDo.Administration.Forms;

[FormScript("Administration.Movie")]
[BasedOnRow(typeof(MovieRow), CheckNames = true)]
public class MovieForm
{
    public string Title { get; set; }
    public string Description { get; set; }
    
    [DisplayName("Cast"), MovieCastEditor, IgnoreName]
    public List<MovieCastRow> CastList { get; set; }
    public string Storyline { get; set; }
    public MovieKind Kind { get; set; }
    public List<int> GenreList { get; set; }
    public string PrimaryImage { get; set; }
    public string GalleryImages { get; set; }
    //[Tab("Movies"), IgnoreName]
    //public string MoviesGrid { get; set; }
    public int Year { get; set; }
    public DateTime ReleaseDate { get; set; }

    [TextAreaEditor(Rows = 3)]
    public int Runtime { get; set; }
}

File: ToDo/ToDo.Web/Modules/ServerTypes/Administration/MovieForm.ts
import { StringEditor, EnumEditor, LookupEditor, ImageUploadEditor, MultipleImageUploadEditor, IntegerEditor, DateEditor, TextAreaEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { MovieCastEditor } from "../../Administration/MovieCast/MovieCastEditor";
import { MovieKind } from "./MovieKind";

export interface MovieForm {
    Title: StringEditor;
    Description: StringEditor;
    CastList: MovieCastEditor;
    Storyline: StringEditor;
    Kind: EnumEditor;
    GenreList: LookupEditor;
    PrimaryImage: ImageUploadEditor;
    GalleryImages: MultipleImageUploadEditor;
    Year: IntegerEditor;
    ReleaseDate: DateEditor;
    Runtime: TextAreaEditor;
}

export class MovieForm extends PrefixedContext {
    static readonly formKey = 'Administration.Movie';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!MovieForm.init)  {
            MovieForm.init = true;

            var w0 = StringEditor;
            var w1 = MovieCastEditor;
            var w2 = EnumEditor;
            var w3 = LookupEditor;
            var w4 = ImageUploadEditor;
            var w5 = MultipleImageUploadEditor;
            var w6 = IntegerEditor;
            var w7 = DateEditor;
            var w8 = TextAreaEditor;

            initFormType(MovieForm, [
                'Title', w0,
                'Description', w0,
                'CastList', w1,
                'Storyline', w0,
                'Kind', w2,
                'GenreList', w3,
                'PrimaryImage', w4,
                'GalleryImages', w5,
                'Year', w6,
                'ReleaseDate', w7,
                'Runtime', w8
            ]);
        }
    }
}

[MovieKind]; // referenced types